pkg/config/mesh/meshwatcher/watcher_test_utils.go

```diff
@@ -0,0 +1,62 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package meshwatcher
```

Comment:
Does it make sense to use the `meshwatcher_test` package here since these test functions/struct names make sense in the context of this package and may be confusing? One other option is require `FixedWatcher` to hold a `*testing.T`

Reply:
You cannot import it then, unfortunately.

Reply:
For tests in other packages? Yeah fair enough; maybe testing.T option would work then to ensure it's only used for tests

Reply:
If we put in a `T` then that import leaks into everything. We can do it but you need to make a separate package only imported in tests which is kind of annoying?

I renamed it TestWatch so I think that should be clear enough now?